### PR TITLE
Use remote window size for send flow control

### PR DESCRIPTION
**Pull request by olix0r**
The initial window size of both the send and recv flow controllers are
configured from the same setting. The send flow controller should be
configured from the remote's initial window size.

I'm unsure of how to test this.

### CODE REVIEW

**Comment by olix0r** [comment]
This isn't a complete fix -- the connection window is not actually set to 654321...
```
1507559240.159027 TRACE h2::proto::streams::recv recv_data; size=9; connection=65535; stream=654321
```

I see this comment in recv.rs, which I don't quite understand:
```

        // connections always have the default window size, regardless of
        // settings
        flow.inc_window(DEFAULT_INITIAL_WINDOW_SIZE)
            .expect("invalid initial remote window size");
        flow.assign_capacity(DEFAULT_INITIAL_WINDOW_SIZE);
```

I suspect that this means that the receive window setting should only be applied once the local peer's setting has been acknowledged by the remote?

**Comment by seanmonstar** [comment]
What you're seeing is according to the spec:

> A SETTINGS frame cannot alter the connection flow-control window.

The connection window can only be changed with WINDOW_UPDATE frames.

File: Cargo.toml [hunk 1]
```diff
@@ -55,3 +55,6 @@ io-dump = { git = "https://github.com/carllerche/io-dump" }
 rustls = "0.11"
 tokio-rustls = { git = "https://github.com/briansmith/tokio-rustls", tag = "b/p1" }
 webpki-roots = "0.13"
+
+[replace]
+"http:0.1.0" = { git = "https://github.com/hyperium/http" }
```
Comment (seanmonstar) [inline Cargo.toml]: This shouldn't be needed any more, we just published 0.1.1 that includes the changes the tests were relying upon.

Comment (hawkw) [reply Cargo.toml]: 🙌

**Comment by seanmonstar** [comment]
> ./ci/h2spec.sh exited with 0.
> Done. Your build exited with 1.

 (╯°□°）╯︵ ┻━┻

**Comment by carllerche** [comment]
Clearly we should be new typing receive vs. send flow values differently...

**Comment by seanmonstar** [comment]
> Clearly we should be new typing receive vs. send flow values differently...

Not sure if joking, but I kinda think it's a good idea. `WindowSize<Local>` and `WindowSize<Remote>` (not exposed externally, just to save ourselves).

**Comment by carllerche** [comment]
It's Ok, nobody ever knows when I'm joking.

**Pull request merged**